net/geth/Makefile

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (C) 2018 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_LICENSE:=ASL-2.0
+PKG_MAINTAINER:=Mislav Novakovic <mislav.novakovic@sartura.hr>
+
+PKG_NAME:=go-ethereum
+PKG_VERSION:=1.8.11
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://codeload.github.com/ethereum/go-ethereum/tar.gz/v${PKG_VERSION}?
+PKG_HASH:=ad18cc1d3154499ade6c712eab4b005d9dc0abf61282cfb349900d30dfba019a
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_USE_MIPS16:=0
```

Comment:
did you confirm that this is required? or did you copy it from somewhere else?

Reply:
I added `PKG_USE_MIPS16:=0` because the package would fail on travis (MIPS, ar7xx), x86 and ARM works just fine. The package is based on https://github.com/openwrt/packages/pull/5903 . I'm still investigating why this is needed and if the binary can be actually run on a MIPS architecture.

Reply:
I was only concerned about the mips16 flag.

Reply:
The `PKG_USE_MIPS16:=0` basically disabled mips 16-bit support for this package and for me that's acceptable.

Reply:
Yes, I know what it does, I was concerned with whether it was _required_ or not.  The option exists because some applications and some toolchains used to miscompile mips16 code for them.  If this is not the case for this package, then it shouldn't be specified.

Reply:
Without `PKG_USE_MIPS16:=0` I see errors like this when I compile the golang, [obfs4proxy](https://github.com/openwrt/packages/pull/5781) or [tor-fw-helper](https://github.com/openwrt/packages/pull/6346) packages:

```
# runtime/cgo
gcc_mipsx.S: Assembler messages:               
gcc_mipsx.S:25: Error: invalid operands `sw $18,12($29)'
gcc_mipsx.S:26: Error: invalid operands `sw $19,16($29)'
gcc_mipsx.S:27: Error: invalid operands `sw $20,20($29)'
...
```

I assume these are the same errors that appear for this package, under the same circumstances. (I haven't tried compiling this package.)

I don't recalling finding a bug report or other source that narrows down the cause, but I assume it's because the assembly language instructions in gcc_mipsx.S were not written with MIPS16 in mind.

To be fair, I'm the one that decided to [enable cgo](https://github.com/openwrt/packages/blob/a81df2bc6108fd1705c03567652914cb246752d0/lang/golang/golang-package.mk#L132). IIRC [cgo is disabled for cross compilation by default](https://golang.org/cmd/cgo/), but Go packages like net and os/user use cgo, so I thought enabling it would maximize package compatibility. (I'm not a real Go developer though so I could be wrong.)

Reply:
Hey, I've got nothing against using the flag, I just don't want it to be cargo culted into every single package :)

Reply:
No worries - I think I'm secretly hoping someone would point out a way to fix these errors so we don't have to disable MIPS16 for every package :laughing:

Reply:
Without `PKG_USE_MIPS16:=0` I get the error:

> make[2]: *** [Makefile:57: /opt/disk/openwrt/openwrt-pi/build_dir/target-mipsel_24kc+24kf_musl/go-ethereum-1.8.11/.built] Error 2
> make[2]: Leaving directory '/opt/disk/openwrt/packages/net/geth'
> Command exited with non-zero status 2

Reply:
@karlp can this be merged now?

Reply:
I'm no gatekeeper, I just ask questions and point out things.